File: python/src/core/reflector.py
"""エニグママシンのリフレクターコンポーネント"""

import string


class Reflector:
    """エニグマリフレクターの実装"""
    
    def __init__(self, mapping):
        """リフレクターを初期化
        
        Args:
            mapping: リフレクターの配線を定義する26文字の文字列
        """
        self.mapping = mapping
    
    def reflect(self, char):
        """リフレクターでの反射"""
        idx = string.ascii_uppercase.index(char)
        return self.mapping[idx]

File: java/src/main/java/com/enigma/core/Reflector.java
package com.enigma.core;

public class Reflector {
    private final String mapping;
    private static final String ALPHABET = "ABCDEFGHIJKLMNOPQRSTUVWXYZ";
    
    public Reflector(String mapping) {
        this.mapping = mapping;
    }
    
    public char reflect(char c) {
        return mapping.charAt(ALPHABET.indexOf(c));
    }
}